src/compiler/emitter.ts

```diff
@@ -5254,6 +5424,17 @@ module ts {
             }
 
             function emitCommonJSModule(node: SourceFile, startIndex: number) {
+                createExternalModuleInfo(node);
+                emitCaptureThisForNodeIfNecessary(node);
+                emitLinesStartingAt(node.statements, startIndex);
+                emitTempDeclarations(/*newLine*/ true);
+                emitExportDefault(node, /*emitAsReturn*/ false);
+            }
+
+            function emitES6Module(node: SourceFile, startIndex: number) {
+                createExternalModuleInfo(node);
+                externalImports = undefined;

```

Comment:
Can we just mark (probably as an argument) that we want to collect only default for the module instead of collecting all data and throwing it away?